### DISCUSSION TITLE
[Question] Can I draw a DEPTH MAP using it?

### DISCUSSION THREAD

**Discussion by exousiayk** [mitsuba-renderer/mitsuba3#1046] [Q&A] [answered]
I'm studying on shoe print for shoe-tread impressions at crime scenes. So, I need to draw a depth map of shoe print. 

How can I draw a depth map in python? I can't find it in Mitsuba Documentation.

![B208420](https://github.com/mitsuba-renderer/mitsuba3/assets/57902052/f45947b5-1d84-419d-b59d-2b39020c52e8)
*sample image

**Comment by njroussel** [comment, marked as answer]
Hi @exousiak 

You'll want to use the [`aov` integrator](https://mitsuba.readthedocs.io/en/latest/src/generated/plugins_integrators.html#arbitrary-output-variables-integrator-aov), with the "depth" attribute.

**Reply by exousiayk** [reply]
Thanks njroussel.

I have a one more question. Is it possible to use AOV input data as a .png or .jpg file instead of .xml?

**Reply by njroussel** [reply]
No ?

Maybe there's a misunderstanding here. The `aov` integrator and its `depth` map simply return the distance from your viewpoint to the shape. This therefore requires that you have a 3D model of your shoe.The XML file only serves a description of your scene and the parameters of the `aov` integrator.

If you're trying to recover the depth map from a single picture (not a 3D shape), you should look into other projects. Mitsuba is not well-suited for that problem.

**Reply by Monoxide-Chen** [reply]
How to write a configuration in an XML file to obtain the depth channel?
The example in the doc is as follows:

```xml
<integrator type="aov">
    <string name="aovs" value="dd.y:depth,nn:sh_normal"/>
    <integrator type="path" name="my_image"/>
</integrator>
```
Is `dd.y:depth` sufficient to extract the depth information?
Do we need the `nn:sh_normal` part and the `<integrator type="path" name="my_image"/>` part, or can they be omitted if we are only interested in the depth channel?

**Reply by njroussel** [reply]
Hi @Monoxide-Chen 
The snippet below is the bare minimum.

```xml
<integrator type="aov">
    <string name="aovs" value="scene_depth:depth"/>
    <integrator type="path" name="my_image"/>
</integrator>
```
